tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/mdspan_to_view_mapper_kk.hpp

```diff
@@ -22,19 +30,72 @@ struct LayoutMapper<std::experimental::layout_right> {
 };
 
 template<class MDSpan>
-std::enable_if_t<MDSpan::rank()==1, Kokkos::View<typename MDSpan::element_type*>>
+std::enable_if_t<
+  MDSpan::rank()==1
+  && !is_complex_v<typename MDSpan::element_type>,
+  Kokkos::View<typename MDSpan::element_type*>
+  >
 mdspan_to_view(const MDSpan& a) {
-    return Kokkos::View<typename MDSpan::element_type*>(a.data(),a.extent(0));
+  return Kokkos::View<typename MDSpan::element_type*>(a.data(), a.extent(0));
 }
 
 template<class MDSpan>
-std::enable_if_t<MDSpan::rank()==2, Kokkos::View<typename MDSpan::element_type**,
-                                                 typename LayoutMapper<typename MDSpan::layout_type>::type>>
-mdspan_to_view(const MDSpan& a) {
-    return Kokkos::View<typename MDSpan::element_type**,
-                        typename LayoutMapper<typename MDSpan::layout_type>::type>
-            (a.data(),a.extent(0),a.extent(1));
+std::enable_if_t<
+  MDSpan::rank()==1
+  && is_complex_v<typename MDSpan::element_type>,
+  Kokkos::View< Kokkos::complex<typename MDSpan::element_type::value_type>* >
+  >
+mdspan_to_view(const MDSpan& a)
+{
+  using mdspan_v_type = typename MDSpan::element_type::value_type;
+  using kc_t = Kokkos::complex<mdspan_v_type>;
+  if (alignof(mdspan_v_type) != alignof(kc_t)){
+    throw std::runtime_error("KokkosBlas: mdspan_to_view: mismatching alignment of std::complex<> and Kokkos::complex<> which prevents interoperability. By default, Kokkos::complex<T> is aligned as 2*sizeof(T) but std::complex<T> is aligned as sizeof(T). To match the alignement and allow correct conversion, you can rebuild Kokkos undefining KOKKOS_ENABLE_COMPLEX_ALIGN.");
```

Comment:
Excellent error message!  Just a couple suggestions:
```suggestion
    throw std::runtime_error("KokkosBlas: mdspan_to_view: mismatching alignment of std::complex<> and Kokkos::complex<> which prevents interoperability. By default, Kokkos::complex<T> is aligned as 2*sizeof(T) but std::complex<T> is aligned as sizeof(T). To match the alignment and allow correct conversion, you must rebuild Kokkos with KOKKOS_ENABLE_COMPLEX_ALIGN undefined.");
```

Reply:
done!